Fix payment gateway suggestions display logic

ilyasfoo opened the pull request:
### Changes proposed in this Pull Request:

This PR
- Fixes payment gateway display logic. Previously, it was displaying "other" gateways for WCPay-ineligible countries even when it's not defined in the "other" category.
- Update tests to reflect the correct logic.
- Added rule to support offline selling venues to `woocommerce_onboarding_profile`
- Added `IN` to PayPal's visibility rule.
- Updates variable names and comments to reduce confusion (hopefully)

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

**Enable core profiler feature**
1. Check out this branch and [build the plugin](https://github.com/woocommerce/woocommerce/blob/trunk/DEVELOPMENT.md).
2. Download and install the [WooCommerce Beta Tester](https://wordpress.org/plugins/woocommerce-beta-tester/) plugin.
3. In `wp-admin`, under **Tools > WCA Test Helper**, click the Features link in the top menu.
4. Find the `core-profiler` feature flag and enable it via the toggle.

**Testing new onboarding option rule**
1. Go to OBW
3. In "Which one of these best describes you?",  Select `I'm already selling` and `No, I'm selling offline`
1. Choose Japan as the store country
1. Complete OBW without installing any plugins in the business step.
1. Go to `WooCommerce > Settings > Advanced > WooCommerce.com` > Untick "Show Suggestions" checkbox
1. Go to `WooCommerce -> Home` and choose Set up payments task.
1. Observe that `Stripe`, `PayPal Payments` and `Square` are displayed in order.

**Testing Paypal in India**
1. Change store country to India
1. Observe that Paypal doesn't show up

<!-- End testing instructions -->

rjchow committed:
add: updated payment gateways for 2023 Q3

ilyasfoo committed:
Update payment gateway suggestions display logic, better comments and variable names, added core onboarding profiler option to support offline venue, added IN to paypal visibility rule

ilyasfoo commented on plugins/woocommerce-admin/client/task-lists/fills/PaymentGatewaySuggestions/utils.js:
```diff
@@ -93,34 +104,34 @@ export const getSplitGateways = (
+						// allow to list "additional" gateways.
+						mainList.push( gateway );
 					}
-				} else if ( ! isWCPaySupported ) {
```
@rjchow This is the actual logic fix. Others are just name fix/update

ilyasfoo committed:
Lint

chihsuan reviewed (approved):
Nice fix! @ilyasfoo  LGTM and tested well. Pre-approving 👍  

There are some lint errors:

```bash
plugins/woocommerce-admin lint: /home/runner/work/woocommerce/woocommerce/plugins/woocommerce-admin/client/task-lists/fills/PaymentGatewaySuggestions/utils.js
plugins/woocommerce-admin lint:    4:1   error  Expected preceding "External dependencies" comment block                                       @woocommerce/dependency-group
plugins/woocommerce-admin lint:    4:10  error  'get' is defined but never used. Allowed unused vars must match /createElement/u               @typescript-eslint/no-unused-vars
plugins/woocommerce-admin lint:    5:10  error  'isWcPaySupported' is defined but never used. Allowed unused vars must match /createElement/u  @typescript-eslint/no-unused-vars
plugins/woocommerce-admin lint:   95:1   error  Missing JSDoc @return type                                                                     jsdoc/require-returns-type
plugins/woocommerce-admin lint: 
``

ilyasfoo committed:
Update for better WCPay logic

ilyasfoo committed:
Remove accidental imports

ilyasfoo committed:
Added return object comment

ilyasfoo commented:
Thanks for the quick review, @chihsuan! I've added a better handling logic for WCPay in ee454a0.

> There are some lint errors:

I've addressed these in 1d20631 and 31fd47b. Thanks for catching that!

ilyasfoo committed:
More lint

ilyasfoo committed:
Merge branch 'trunk' into fix/payment-gateway-suggestions-logic

ilyasfoo committed:
Changelog
